addons/hr_calendar/models/calendar_event.py

```diff
@@ -2,34 +2,31 @@
 from dateutil.relativedelta import relativedelta
 from pytz import UTC
 
-from odoo import api, fields, models
+from odoo import api, models
 
 from odoo.addons.resource.models.utils import Intervals, sum_intervals, timezone_datetime
 
 
 class CalendarEvent(models.Model):
     _inherit = "calendar.event"
 
-    unavailable_partner_ids = fields.Many2many('res.partner', compute='_compute_unavailable_partner_ids')
-
-    @api.depends('partner_ids', 'start', 'stop', 'allday')
+    @api.depends('allday')
     def _compute_unavailable_partner_ids(self):
+        super()._compute_unavailable_partner_ids()
         complete_events = self.filtered(
             lambda event: event.start and event.stop and (event.stop > event.start or (event.stop >= event.start and event.allday)) and event.partner_ids)
-        incomplete_event = self - complete_events
-        incomplete_event.unavailable_partner_ids = []
         if not complete_events:
             return
         event_intervals = complete_events._get_events_interval()
         for event, event_interval in event_intervals.items():
             # Event_interval is empty when an allday event contains at least one day where the company is closed
             if not event_interval:
-                event.unavailable_partner_ids = event.partner_ids
+                event.unavailable_partner_ids |= event.partner_ids
```

Comment:
Isn't that part already covered by the base calendar logic?

Normally the only addition of the hr override is to take the working calendar of the employees into account.